### PR TITLE
Using UInt16 view instead of dealing with bitwise operators

### PR DESCRIPTION
**Pull request by WebReflection**
This MR drops the unnecessary dance around char codes and simply uses *UInt16Array* as suggested.

### CODE REVIEW

File: cjs/index.js [hunk 1]
```diff
@@ -23,28 +23,24 @@ module.exports = (self, {parse, stringify} = JSON) => {
...
         self.postMessage({[CHANNEL]: {action, args, id, sb}});
         Atomics.wait(i32a, 0);
 
         // commit transaction using the right buffer length
-        const length32 = Atomics.load(i32a, 0);
-        const length16 = ceil(length32 / 2);
-        sb = new SharedArrayBuffer(length16 * BYTES_PER_ELEMENT);
+        const length = Atomics.load(i32a, 0);
+        const bytes = UI16_BYTES * length;
+        sb = new SharedArrayBuffer(bytes + (bytes % 4));
```
Comment (hoodmane) [inline cjs/index.js:34]:
```suggestion
        sb = new SharedArrayBuffer(bytes - (((-bytes)%4 + 4) % 4));
```

**Comment by hoodmane** [comment]
You want to round bytes up to a multiple of `I32_BYTES`. You can round down with `bytes & I32_BYTES`. I guess the shortest way to round up is:
```
(bytes + I32_BYTES - 1) & I32_BYTES
```

**Comment by hoodmane** [comment]
Of the three options, `Math.ceil` is the most readable:
```js
bytes + (((-bytes)%I32_BYTES + I32_BYTES) % I32_BYTES)
(bytes + I32_BYTES - 1) & (~(I32_BYTES-1))
Math.ceil(bytes/I32_BYTES) * I32_BYTES
```

**Comment by hoodmane** [comment]
Oh I get it you have a number that is already divisible by 2, so `bytes + (bytes % 4)` is rounding up to the nearest multiple of 4.

**Comment by WebReflection** [comment]
> Oh I get it you have a number that is already divisible by 2, so bytes + (bytes % 4) is rounding up to the nearest multiple of 4.

yeah, the length is the UTF-16 length of the serialized string already, I just need to reach *next 4 bytes* there, and I think it's working just fine now ... ?

**Comment by WebReflection** [comment]
OK, I am going to merge this ... if there's any use case you have in mind that would/should fail please let me know, thank you @hoodmane !!!

**Pull request merged**